vmdb/app/controllers/provider_foreman_controller.rb

```diff
@@ -673,10 +693,32 @@ def construct_edit
                    :verify_ssl => params[:verify_ssl]}
   end
 
+  def locals_for_tagging
+    {:action_url   => 'tagging',
+     :multi_record => true,
+     :record_id    => @sb[:rec_id] || @edit[:object_ids] && @edit[:object_ids][0]
+    }
+  end
+
+  def update_tagging_partials(presenter, r)
+    presenter[:update_partials][:main_div] = r[:partial => 'layouts/tagging',
+                                               :locals  => locals_for_tagging]
+    presenter[:update_partials][:form_buttons_div] = r[:partial => 'layouts/x_edit_buttons',
+                                                       :locals  => locals_for_tagging]
+  end
+
+  def clear_flash_msg
+    @flash_array = nil if params[:button] != "reset"
+  end
+
   def breadcrumb_name
     ui_lookup_for_model(self.class.model_name).singularize
   end
 
+  def tagging_explorer_controller?
+    @explorer && true
```

Comment:
This is weird.  What are you trying to do here?  If you're trying to get a literal boolean, then just do `!!@explorer`.  However most of the time that is not necessary at all, and this method could just return `@explorer`